assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -658,57 +675,81 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
+      // If scaleFactor has already been applied to the line-map, we need to reverse that for calculations
+      const unscaledHeight = lineDiagramHeight / scaleFactor;
+      const unscaledWidth = lineDiagramWidth / scaleFactor;
 
       // First, scale x. Then, check if it needs abbreviating. Then scale y, given the abbreviation
-      let xScaleFactor = 904 / unscaledWidth;
+      let xScaleFactor = fullWidth / unscaledWidth;
       
       const needsAbbreviating = !doAbbreviate &&
-        unscaledHeight * xScaleFactor + getEmphasisHeight(xScaleFactor) > svgHeight;
+        unscaledHeight * xScaleFactor + getEmphasisHeight(xScaleFactor) > diagramContainerHeight;
       if (needsAbbreviating) {
         setDoAbbreviate(true);
         // now scale y, which requires re-running this effect
       } else {
-        const yScaleFactor = (svgHeight - getEmphasisHeight(1)) / unscaledHeight
+        const yScaleFactor = (diagramContainerHeight - getEmphasisHeight(1)) / unscaledHeight
         const factor = Math.min(
           xScaleFactor,
           yScaleFactor
         );
         setScaleFactor(factor);
-        setTimeout(() => {
-          setIsDone(true);
-        }, 200);
+        setIsDone(true);
       }
     }
   }
 
   // When the parent container size changes, or abbreviation setting changes,
-  // re-measure the diagram and scale accordingly
+  // re-measure the diagram and scale accordingly.
   useEffect(() => {
-    setTimeout(() => {
-      measureDiagramAndScale()
-    }, 100)
-  }, [svgHeight, doAbbreviate]);
+    // The isCurrent setting is needed to clean up the unused hook runs / state changes
+    let isCurrent = true
+    
+    // The document.fonts.ready.then() is needed when the font takes a while to load
+    // but the diagram measurements have already been taken.
+    // Example: shuttle Chinatown > Mass Ave, screen located at Back Bay
+    document.fonts.ready.then(() => {
+      if (isCurrent) measureDiagramAndScale()
+    })
+    return () => {
+      isCurrent = false
+    }
+  }, [lineDiagramHeight, diagramContainerHeight, doAbbreviate]);
 
   // This is to center the diagram along the X axis
-  const translateX = (width && (904 - width) / 2) || 0;
+  const translateX = (lineDiagramWidth && (fullWidth - lineDiagramWidth) / 2) || 0;
   
   // Next is to align the diagram at the top of the svg, which involves adjusting the SVG viewbox
 
+  // First, we need to figure out whether we're in a Screenplay simulation or not,
+  // because unfortunately, the CSS transform on those simulations messes up the widget's
+  // ability to measure itself in the DOM. So, we need to get the original height of the
+  // diagram, pre-scaled, to accurately set its viewbox dimensions.
+  const simulation = document.getElementById("simulation")
+  const simulationStyle = simulation && window.getComputedStyle(simulation) 
+  const simulationTransform = new WebKitCSSMatrix(simulationStyle?.transform).m11;
```

Comment:
[It looks like `DOMMatrix` is the more standard/general name for this guy](https://developer.mozilla.org/en-US/docs/Web/API/DOMMatrix), does it work if you switch to that?